### DISCUSSION TITLE
GPT-3 responds in place of GPT-4

### DISCUSSION THREAD

**Discussion by ResielCWS** [openai-php/client#165] [General]
While requiring the use of the GPT-4 model, when I ask the AI to identify itself with name and version, it emerges that it is version 3. Why?

**Comment by ResielCWS** [comment]
even requesting the use of the "gpt-4-0613" model, the gpt-3 model actually responds

**Comment by pb30** [comment]
GPT is not an expert on itself. It was trained on material before it existed. You can search OpenAI forums for more threads on this topic. If it’s critical for it to respond correctly in the text, add this fact to your system prompt (likely what ChatGPT does). Otherwise look at the model attribute in the response.

**Comment by ResielCWS** [comment]
If I ask GPT-3.5 from chat.openai.com "User
Identify yourself: AI name and version" replies: "I am ChatGPT, an AI language model developed by OpenAI. I am based on the GPT-3.5 architecture. How may I assist you today?”.

If I ask GPT-4 the same thing he replies: "I am ChatGPT, an artificial intelligence developed by OpenAI. Specifically, I am based on the GPT-4 architecture. My purpose is to provide assistance, information, and perform tasks involving natural language understanding and generation. It's important to note that as of my training cut-off in September 2021, I am unable to access real-time data or learn new information after that date"

It seems to me that he identifies very well. The problem is that when I ask the GPT-4 model with this PHP client it replies identifying itself as GPT-3. This doesn't surprise me: because it makes the same mistakes as GPT-3

**Comment by pb30** [comment]
As I said, chat.openai.com likely includes this information in the system message which we have no visibility into, and you can do the same if it is important to you:

```php
$response = $client->chat()->create([
    "model"    => "gpt-4",
    "messages" => [
      [
        "role"    => "system",
        "content" => "You are a helpful assistant, powered by GPT-4 by OpenAI.",
      ],
      [
        "role"    => "user",
        "content" => "Who are you?",
      ],
    ],
  ]);
```

> [content] => I'm a digital assistant powered by GPT-4, which is a language-based AI model developed by OpenAI. I exist to help users like you with information, answers, and tasks. I don't have personal experiences or emotions. Instead, I analyze inputs based on a large dataset of human language and generate responses accordingly. Please note that I don't have the ability to access personal data about individuals unless it has been shared with me in the course of our conversation. I am designed to respect user privacy and confidentiality.

You can try asking the same thing in the [OpenAI Playground](https://platform.openai.com/playground?mode=chat&model=gpt-4) to see it is not an issue with this PHP library.
